### PR TITLE
Update documentation section related to tests

### PR DESCRIPTION
**Pull request by kersommoura**
Add info about functional tests, and the use of Pulp smash to write functional tests.

### CODE REVIEW

File: docs/contributing/continuous-integration.rst [hunk 1]
```diff
@@ -1,29 +1,44 @@
...
 
-New code is highly encouraged to have basic unit tests that demonstrate its functionality. A Pull
-Request that has failing unit tests cannot be merged.
+Unit Tests
+----------
+New code is highly encouraged to have basic unit tests that demonstrate that
+units (function, method or class instance) are working correctly.
+A Pull Request that has failing unit tests cannot be merged.
+
+The unit tests for `pulpcore` are in `pulpcore/tests <https://github.com/pulp/pulp/tree/master/pulpcore/tests/units>`_.
```
Comment (gmbnomis) [inline docs/contributing/continuous-integration.rst]: The link should be https://github.com/pulp/pulp/tree/master/pulpcore/tests/unit (no 's' at the end)

File: docs/contributing/continuous-integration.rst [hunk 2]
```diff
@@ -1,29 +1,44 @@
...
+
```
Comment (gmbnomis) [inline docs/contributing/continuous-integration.rst]: As of today, there are also pulpcore-plugin unit tests in https://github.com/pulp/pulp/tree/master/plugin/tests/unit/. We might want to mention the possibility to add plugin API related unit tests as well.

File: docs/contributing/continuous-integration.rst [hunk 3]
```diff
@@ -1,29 +1,44 @@
...
+Functional Tests
+----------------
+
+Functional tests verify a specific action or feature.
```
Comment (bmbouter) [inline docs/contributing/continuous-integration.rst]: Is it helpful to mention that Pulp never releases with failing unit or functional test failures. So contributing test failures is a great way to ensure that your workflows never regress. It's kind of like our user-contract that they can help define. I'm also ok to leave this out, whatever you want to do.

**Comment by bmbouter** [comment]
@kersommoura the docs are failing though, so Travis needs to pass first (obviously).

File: docs/contributing/continuous-integration.rst (hunk 1 above)
Comment (kersommoura) [reply docs/contributing/continuous-integration.rst]: Thanks.

File: docs/contributing/continuous-integration.rst (hunk 2 above)
Comment (kersommoura) [reply docs/contributing/continuous-integration.rst]: Done.

**Comment by kersommoura** [comment]
@bmbouter, I updated per your suggestion, and travis is happy now.

File: docs/contributing/continuous-integration.rst [hunk 4]
```diff
@@ -1,32 +1,54 @@
...
+----------------
 
-The unit tests for `pulpcore` are in `pulpcore/tests <https://github.com/pulp/pulp/tree/master/pulpcore/tests>`_.
+Functional tests verify a specific action or feature.
+In general functional tests tend to answer the question "As an user can I do this?"
 
-Integration tests for new code should be added to a separate project called pulp-smash_. A Pull
-Request that has failing pulp-smash tests cannot be merged.
+Functional tests for Pulp are written using pulp-smash_ . Pulp smash is a test
+framework written to ease the test of Pulp.
```
Comment (dkliban) [inline docs/contributing/continuous-integration.rst]: s/ease the test/ease the testing/

File: docs/contributing/continuous-integration.rst [hunk 5]
```diff
@@ -1,32 +1,54 @@
...
 
-The integration tests for the REST API live in the pulpcore_ folder and the Plugin API is tested
-with tests in the file_ folder.
+It is highly encouraged to new features to be tested adding a functional test
```
Comment (dkliban) [inline docs/contributing/continuous-integration.rst]: It is highly encouraged to accompany new features with functional tests in `pulpcore/functional <https://github.com/pulp/pulp/tree/master/pulpcore/tests/functional>`_.

File: docs/contributing/continuous-integration.rst [hunk 6]
```diff
@@ -1,32 +1,54 @@
...
+to `pulpcore/functional <https://github.com/pulp/pulp/tree/master/pulpcore/tests/functional>`_.
 
+Only tests for features related to `pulpcore` should live on the previous path.
```
Comment (dkliban) [inline docs/contributing/continuous-integration.rst]: Only the tests tests for features related to `pulpcore` should live in this repository.

File: docs/contributing/continuous-integration.rst [hunk 7]
```diff
@@ -1,32 +1,54 @@
...
+<https://github.com/pulp/pulp/tree/master/pulpcore/tests/unit>`_.
+
+The unit tests for `pulpcore-plugin` are in `pulpcore-plugins/tests
+<https://github.com/pulp/pulp/tree/master/plugin/tests/unit/>`_.
+
+Functional Tests
+----------------
 
-The unit tests for `pulpcore` are in `pulpcore/tests <https://github.com/pulp/pulp/tree/master/pulpcore/tests>`_.
+Functional tests verify a specific action or feature.
```
Comment (dkliban) [inline docs/contributing/continuous-integration.rst]: s/action or feature/feature/

File: docs/contributing/continuous-integration.rst [hunk 8]
```diff
@@ -37,3 +59,10 @@ Attention and care must be given to merging PRs that require other Pull Requests
 all required PRs should be ready to merge--meaning that all tests/checks should be passing, the code
 review requirements should be met, etc. When merging, the PR along with its required PRs should all
 be merged at the same time. This is necessary to ensure that test breakages don't block other PRs.
+
+Contributing to tests
+----------------------
+
+Pulp team will never release a new Pulp version with failing units or
```
Comment (dkliban) [inline docs/contributing/continuous-integration.rst]: A new version of Pulp will only be released when all unit and functional tests are passing. Contributing tests is a great way to ensure that your workflows never regress.

**Comment by kersommoura** [comment]
@dkliban, thanks for your review. Your comments were addressed.

**Pull request merged**